support/modules-artifact-gen/single-file-artifact-gen

```diff
@@ -155,8 +155,14 @@ echo "$dest_dir" > $dest_dir_file
 # Create single_file file in plain text
 echo "$filename" > $filename_file
 
+set +e;
+STAT_HAS_f=0;
+stat -f %A "${file}" >/dev/null 2>&1;
+[ $? -eq 0 ] && STAT_HAS_f=1;
+set -e;
+
 # Create permissions file in plain text
-if [[ "$OSTYPE" == "darwin"* ]]; then
+if [ "$STAT_HAS_f" == "1" ]; then
   stat -f %A "${file}" > $permissions_file
```

Comment:
does the existence of the `-f` flag also imply the `%A`?

Reply:
> does the existence of the `-f` flag also imply the `%A`?

I do not know:) but look, it check if the command was successful. if `$?==0` then %A was supported. do you think we should add checking if output matched `^[0-9]+$`?

Reply:
na, I just saw that on some example file:
```
stat -f %a LICENSE 
1565790353
stat -f %A LICENSE
644
```
the generation will fail... so the additional check would be helpful, although in my opinion it should be on the person modifying their OS to live with the consequences 😜

Reply:
> na, I just saw that on some example file:
> 
> ```
> stat -f %a LICENSE 
> 1565790353
> stat -f %A LICENSE
> 644
> ```
> 
> the generation will fail... so the additional check would be helpful, although in my opinion it should be on the person modifying their OS to live with the consequences 😜

lots of sense here as well. check if you like [it](https://github.com/mendersoftware/mender/pull/428/commits/2ca77cc03487655e5f25525df2e1fb1518fc3e41)